src/query/execute_statement.c

```diff
@@ -14807,14 +14809,74 @@ do_replicate_statement (PARSER_CONTEXT * parser, PT_NODE * statement)
     }
   else
     {
-      PT_PRINT_VALUE_FUNC saved_func = parser->print_db_value;
-      int saved_custom_print = parser->custom_print;
+      /*
+       * if the query string includes the host variables, while processing the variable holder '?'
+       * the values of the host variables can be replaced into the user's original query string
+       * the pt_print_db_value(...) returns the value string and its length.
+       * the length includes quotes in case of the char string.
+       */
+      char *sql_text = statement->sql_user_text;
+      int sql_len = statement->sql_user_text_len;
+      int i, n, nth;
+      int var_len = 0;
+      bool begin_quote = false;
+
+      host_val = (PARSER_VARCHAR **) malloc (sizeof (PARSER_VARCHAR *) * parser->host_var_count);
+      if (host_val == NULL)
+	{
+	  return ER_OUT_OF_VIRTUAL_MEMORY;
+	}
+
+      for (i = 0; i < parser->host_var_count; i++)
+	{
+	  host_val[i] = pt_print_db_value (parser, &parser->host_variables[i]);
+	  var_len += host_val[i]->length;
+	}
+
+      sbr_text = (char *) malloc (sql_len + var_len);
+      if (sbr_text == NULL)
+	{
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto end;
+	}
+
+      n = nth = 0;
+
+      for (i = 0; i < sql_len; i++)
+	{
+	  if (sql_text[i] == '\'')
+	    {
+	      if (!begin_quote)
```

Comment:
As far as I know, CUBRID uses '' instead of \' to represent quotation marks, so this shouldn't be a problem. I am just worried that there may be a problem related to the quotation marks in the string. So I leave a comment hoping you will confirm. (I don't find any bad case for this.)

Reply:
I checked the case which you said as your comment but it was no problem. thanks.